jobsdb/jobsdb.go

```diff
@@ -1412,11 +1414,16 @@ func computeInsertVals(before, after []string) ([]string, error) {
 				return calculatedVals, fmt.Errorf("computed index is invalid. before: %v, after: %v, calculatedVals: %v", before, after, calculatedVals)
 			}
 		}
-
+		var comparatorBool bool
+		if skipZeroAssertionForMultitenant {
+			comparatorBool = len(calculatedVals) == 2
+		} else {
+			comparatorBool = (before[0] == "0" && len(calculatedVals) == 3) ||
+				(before[0] != "0" && len(calculatedVals) == 2)
```

Comment:
I think this could lead to a `panic` given that up above we are only checking if `before` lenght is `> 4` but never if it has at least one element. I understand this is called by `computeInsertIdx` only at the moment but every function is a "unit" and at the moment it looks like this unit can break?

```go
func TestPanic(t *testing.T) {
	_, _ = computeInsertVals(nil, nil)
}
```

Gives:

```
=== RUN   TestPanic
--- FAIL: TestPanic (0.00s)
panic: runtime error: index out of range [-1] [recovered]
	panic: runtime error: index out of range [-1]

goroutine 85 [running]:
testing.tRunner.func1.2({0x11b92a0, 0xc000122fa8})
	/usr/local/go/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1392 +0x39f
panic({0x11b92a0, 0xc000122fa8})
	/usr/local/go/src/runtime/panic.go:838 +0x207
github.com/rudderlabs/rudder-server/jobsdb.computeInsertVals({0x0?, 0x51e9b4?, 0x0?}, {0x0?, 0x0, 0x0})
	/home/francesco/Code/rudderstack/rudder-server/jobsdb/jobsdb.go:1392 +0x639
github.com/rudderlabs/rudder-server/jobsdb.TestPanic(0x0?)
	/home/francesco/Code/rudderstack/rudder-server/jobsdb/jobsdb_test.go:26 +0x29
testing.tRunner(0xc000503520, 0x12de2d0)
	/usr/local/go/src/testing/testing.go:1439 +0x102
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1486 +0x35f


Process finished with the exit code 1
```

It should instead return an error instead of killing the whole service. WDYT?

Also, it would be nice to see some fuzzy testing or if that's too much for now just test the same unit with bad data to make sure it doesn't break.

Reply:
Added a check to ensure neither of the inputs is a nil array. Probably should add more validations to ensure we are not panicking

Reply:
That is great, thanks @cisse21 :+1: 
I think we just need to cover them now

![Screenshot from 2022-05-26 13-56-55](https://user-images.githubusercontent.com/1288256/170483656-23ecf5fa-1aee-4336-9dc6-9b59c847f858.png)

Reply:
I will add tests for these cases too